collector/bgp.go

```diff
@@ -126,54 +149,7 @@ func processBgpL2vpnEvpnSummary(ch chan<- prometheus.Metric, jsonBGPL2vpnEvpnSum
 	return nil
 }
 
-// Helper function to set Prometheus metrics descriptions. BGPv4 metrics can have optional
-// labels (i.e. peer, peer_as, peer_desc) as defined via the --collector.bgp.peer-descriptions flag,
-// so it's not possible to set the descriptions via a global var.
-func setBgpDesc() {
-	if bgpDesc != nil {
-		return
-	}
-	bgpDescMtx.Lock()
-	defer bgpDescMtx.Unlock()
-	bgpLabels := []string{"vrf", "afi", "safi", "local_as"}
-	bgpPeerTypeLabels := []string{"type", "afi", "safi"}
-	bgpPeerLabels := append(bgpLabels, "peer", "peer_as")
-
-	if *bgpPeerDescs {
-		bgpPeerLabels = append(bgpLabels, "peer", "peer_as", "peer_desc")
-	}
-	bgpPeerMetricPrefix := "bgp_peer"
-
-	bgpDesc = map[string]*prometheus.Desc{
-		"ribCount":        colPromDesc(bgpSubsystem, "rib_count_total", "Number of routes in the RIB.", bgpLabels),
-		"ribMemory":       colPromDesc(bgpSubsystem, "rib_memory_bytes", "Memory consumbed by the RIB.", bgpLabels),
-		"peerCount":       colPromDesc(bgpSubsystem, "peers_count_total", "Number peers configured.", bgpLabels),
-		"peerMemory":      colPromDesc(bgpSubsystem, "peers_memory_bytes", "Memory consumed by peers.", bgpLabels),
-		"peerGroupCount":  colPromDesc(bgpSubsystem, "peer_groups_count_total", "Number of peer groups configured.", bgpLabels),
-		"peerGroupMemory": colPromDesc(bgpSubsystem, "peer_groups_memory_bytes", "Memory consumed by peer groups.", bgpLabels),
-
-		"msgRcvd":               colPromDesc(bgpPeerMetricPrefix, "message_received_total", "Number of received messages.", bgpPeerLabels),
```

Comment:
Why not
```
colPromDesc(bgpSubsystem, "peer_message_received_total", [...]
```
?

Reply:
I am using another variable, `bgpPeerMetricPrefix`, for convenience as there are 7 peer metrics. Is there an advantage of doing:

```
colPromDesc(bgpSubsystem, "peer_message_received_total", [...]
colPromDesc(bgpSubsystem, "peer_message_sent_total", [...]
...
```

rather than:
```
bgpPeerMetricPrefix := "bgp_peer"
colPromDesc(bgpPeerMetricPrefix, "peer_message_received_total", [...]
colPromDesc(bgpPeerMetricPrefix, "peer_message_sent_total", [...]
...
?

Reply:
It's purely for consistency, since you also have four metrics like `peerGroupCount` compared to these 7 peer metrics.